### DISCUSSION TITLE
How do I "List" the installed libraries with a specific triplet

### DISCUSSION THREAD

**Discussion by gizmocuz** [microsoft/vcpkg#17905] [Ideas]
I searched the issues but could not find the answer.

The documentation does not mention it as well

https://docs.microsoft.com/en-us/cpp/build/vcpkg

Since I installed libraries for multiple triplets I would like to see a list for a specific triplet

I tried:

`vcpkg --triplet x86-windows list`

But it displays the packages for all triplets

```
zlib:x64-windows                                   1.2.11-6         A compression library
zlib:x64-windows-static                            1.2.11-6         A compression library
zlib:x86-windows                                   1.2.11-6         A compression library
zlib:x86-windows-static                            1.2.11-6         A compression library
```

Is this feature implemented ? If not, it would be great to have

**Comment by linquize** [comment]
In the mean time you can run `vcpkg list | grep x86-windows`

**Comment by gizmocuz** [comment]
Thanks for looking into this.

Could I propose some additional features:

- Detection of missing library in a triplet
For instance, zlib, installed in x86-windows, but not found in x64-windows

```
zlib:x64-windows                                   1.2.11-6         A compression library
zlib:x64-windows-static                            >> Not Installed <<
```
_(It would be great is 'install' would detect the used triplets on a system and installs for all triplets)_

- Detection of different versions
For instance zlib 1.2.11-5 in x86-windows, 1.2.11-6 in x64-windows

- Minimize output
When zlib is installed in all triplets, only display

```
zlib:all                                             1.2.11-6         A compression library
```

**Comment by Farwaykorse** [comment]
> Thanks for looking into this.
> 
> Could I propose some additional features:
> [...]

That might be tricky since vcpkg allows for custom and now [community triplets][community].
You'll probably never see `all` and the missing list will be really long.

[community]: https://github.com/microsoft/vcpkg/tree/master/triplets/community

**Comment by DimRochette** [comment]
Same issue when I tried to do `vcpkg list --triplet x64-windows`

The ugly workaound on windows is

`vcpkg list | find /i "x64-windows"`

**Comment by JackBoosY** [comment]
Please use command `./vcpkg list TRIPLET_NAME`.

Edit: But we did not completely compare the input parameter.

**Reply by playgithub** [reply]
The syntax `./vcpkg list --triplet=TRIPLET_NAME` is more consistent.
